File: backend/common/src/main/java/afsdigital/grahamselect/valuation/domain/entities/GrahamRecommendation.java
package afsdigital.grahamselect.valuation.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GrahamRecommendation {

    private UUID id;
    private String userId;
    private String ticker;
    private BigDecimal currentPrice;
    private BigDecimal intrinsicValue;
    private BigDecimal marginOfSafety;
    private BigDecimal currentAllocationPct;
    private BigDecimal targetAllocationPct;
    private BigDecimal allocationGap;
    private BigDecimal recommendationScore;
    private LocalDate generatedAt;

}

File: frontend/lib/src/features/ranking/domain/entities/graham_recommendation.dart
class GrahamRecommendation {
  final String ticker;
  final double currentPrice;
  final double intrinsicValue;
  final double marginOfSafety;
  final double currentAllocationPct;
  final double targetAllocationPct;
  final double allocationGap;
  final double recommendationScore;
  final double? epsUsed;
  final double? bvpsUsed;

  const GrahamRecommendation({
    required this.ticker,
    required this.currentPrice,
    required this.intrinsicValue,
    required this.marginOfSafety,
    required this.currentAllocationPct,
    required this.targetAllocationPct,
    required this.allocationGap,
    required this.recommendationScore,
    this.epsUsed,
    this.bvpsUsed,
  });
}
